tempesta_fw/cfg.c

```diff
@@ -747,27 +747,51 @@ spec_start_handling(TfwCfgSpec specs[])
 		if (spec->handler == &tfw_cfg_handle_children)
 			BUG_ON(!spec->cleanup);
 		spec->__called_now = false;
+		if (parent && spec->allow_reconfig && !parent->allow_reconfig) {
+			spec->allow_reconfig = false;
+			TFW_ERR_NL("Reconfiguration ability for directive '%s' "
+				   "is disabled: parent directive '%s' "
+				   "doesn't allow reconfiguration!\n",
+				   spec->name, parent->name);
+		}
 	}
 }
 
+static void
+spec_start_handling(TfwCfgSpec specs[])
+{
+	__spec_start_handling(NULL, specs);
```

Comment:
Just a thought here.
So, an error in `specs[]` is found, where `spec->allow_reconfig` is set whereas the parent directive doesn't allow reconfiguration. Shouldn't this be a real error? This looks like a coding error (not the external configuration file error). So why this coding error is just corrected on the fly instead of hard stopping and allowing the developer to fix it right away?

Reply:
I was too concerned about reconfiguration and not breaking running system that i've missed that the check will happen on the very first start, whatever the configuration would be.

Reply:
Replaced the error report with BUG_ON. Since we have tests that use directives marked as `allow_reconfig` the coding error won't make it's way to production. 
Other such directives should be thoroughly tested since BUG_ON may fire during reconfiguration of running system. That happens if the affected directive is added to the configuration only during reconfiguration and not present in startup configuration.